src/containers/my-quizzes/quiz-settings-container/QuizSettingsContainer.tsx

```diff
@@ -53,74 +47,55 @@ const QuizSettingsContainer = ({
   const { t } = useTranslation()
   const { id } = useParams()
   const navigate = useNavigate()
-  const dispatch = useAppDispatch()
+  const { handleErrorAlert, handleAlert } = useSnackbarAlert()
 
-  const editQuiz = useCallback(
-    (params?: UpdateQuizParams) => ResourceService.editQuiz(params),
-    []
+  const editQuizService = useCallback(
+    async (data: UpdateQuizParams) => {
+      if (id) {
+        await ResourceService.editQuiz(data)
+      }
+    },
+    [id]
   )
 
   const createQuizService = useCallback(
-    (data?: CreateQuizParams) => ResourceService.addQuiz(data),
+    (data: CreateQuizParams) => ResourceService.addQuiz(data),
     []
   )
 
-  const onResponse = () => {
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.success,
-        message: id
-          ? 'myResourcesPage.quizzes.successEditedQuiz'
-          : 'myResourcesPage.quizzes.successAddedQuiz'
-      })
-    )
+  const handleResponse = () => {
+    handleAlert({
+      severity: snackbarVariants.success,
+      message: id
+        ? 'myResourcesPage.quizzes.successEditedQuiz'
+        : 'myResourcesPage.quizzes.successAddedQuiz'
+    })
 
     id
       ? setActiveTab(QuizTabsEnum.Edit)
       : navigate(authRoutes.myResources.root.path)
   }
 
-  const onResponseError = (error?: ErrorResponse) => {
-    const errorKey = getErrorKey(error)
-
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.error,
-        message: error
-          ? {
-              text: errorKey,
-              options: {
-                message: getErrorMessage(error.message)
-              }
-            }
-          : errorKey
-      })
-    )
-  }
-
-  const { fetchData: updateQuiz } = useAxios<null, UpdateQuizParams>({
-    service: editQuiz,
-    fetchOnMount: false,
-    defaultResponse: null,
-    onResponse,
-    onResponseError
+  const { mutate: fetchAddQuiz } = useMutation({
+    mutationFn: createQuizService,
+    onSuccess: handleResponse,
+    onError: handleErrorAlert
   })
 
-  const { fetchData: createQuiz } = useAxios<Quiz, CreateQuizParams>({
-    service: createQuizService,
-    fetchOnMount: false,
-    defaultResponse: { ...defaultResponse, id: '' },
-    onResponse,
-    onResponseError
+  const { mutate: fetchEditedQuiz } = useMutation({
```

Comment:
Just make names more meaningful (edit or update or something like that)
```suggestion
  const { mutate: editQuiz } = useMutation({
```